core/src/autogluon/core/task/base/base_task.py

```diff
@@ -1,172 +1,26 @@
-import collections
 import copy
-import logging
-import time
-from abc import abstractmethod
 
-from ...scheduler import HyperbandScheduler, RLScheduler, FIFOScheduler
+from ...scheduler import HyperbandScheduler, FIFOScheduler
 from ...scheduler.seq_scheduler import LocalSequentialScheduler
-from ...utils import in_ipynb, try_import_mxnet
 from ...utils.utils import setup_compute
 
 __all__ = [
-    'BaseTask',
-    'compile_scheduler_options',
-    'compile_scheduler_options_v2',
-    'create_scheduler']
-
-Results = collections.namedtuple('Results', 'model reward config time metadata')
+    'compile_scheduler_options_v2']
 
 schedulers = {
     'local': LocalSequentialScheduler,
     'fifo': FIFOScheduler,
-    'rl': RLScheduler,
     'hyperband_stopping': HyperbandScheduler,
     'hyperband_promotion': HyperbandScheduler,
 }
 
-logger = logging.getLogger(__name__)
-logger.setLevel(logging.WARNING)
-
-def create_scheduler(train_fn, scheduler, scheduler_options):
-    if isinstance(scheduler, str):
-        scheduler_cls = schedulers[scheduler.lower()]
-    else:
-        assert callable(scheduler)
-        scheduler_cls = scheduler
-        scheduler_options = copy.copy(scheduler_options)
-    return scheduler_cls(train_fn, **scheduler_options)
-
-
-class BaseTask(object):
```

Comment:
@yinweisu This code is never used in AG, but it is [used in GluonCV](https://github.com/dmlc/gluon-cv/blob/master/gluoncv/auto/tasks/image_classification.py#L21). We may need to move this code to GluonCV and do a release as otherwise we can't get rid of it in AG. There are a variety of other problems with this code being used in GluonCV, since this also uses the schedulers/searchers which we are planning to get rid of in v0.4 in favor of Syne Tune / Ray Tune.

Reply:
Essentially we need to either

1. refactor GluonCV to not depend on AG (and also refactor AG.vision to avoid calling gluoncv.auto) -> My preferred option
2. AG drops support for GluonCV and we use a different vision backend -> Preferred option if removing AG from GluonCV is too difficult / conflicts with existing GluonCV users
3. refactor GluonCV and AG to work similarly to how they do now but with the updated scheduler / searcher API and continue having GluonCV depend on AG. (**Not ideal**)